vn/babel upgrade

vickingan opened the pull request.

sjzavaleta commented on es6-lib/upstream/core.js:
```diff
@@ -127,7 +125,7 @@ class Core extends CoreClient {
 
   getView(fourfour, onComplete, retried) {
-    const doGetView = () => this.getView(layer, onComplete, true);
+    const doGetView = () => this.getView(fourfour, onComplete, true);
```
This seems like a big change, but I don't know what the implications are

vickingan replied:
This should be a typo. In the scope of this function, `layer` does not exist, only `fourfour`. This line is also calling `this.getView(...)` which refers to the current function.

sjzavaleta reviewed (commented):
I suppose the updated compiler required a ton of formatting changes. It was sort of hard to tell if you made any changes that weren't prompted by that.

vickingan commented on docker/README.md:
````diff
@@ -6,7 +6,7 @@
 To build the image, run:
 ```
-cp -r ../lib ../scripts ../package.json ../npm-shrinkwrap.json .
+cp -r ../lib ../package.json .
````
It doesn't. I was testing docker as part of this upgrade and the README is out of date. There are no scripts and no npm-shrinkwrap in this repo.

vickingan commented on es6-lib/config/index.js:
```diff
@@ -11,17 +11,15 @@
 import _ from 'underscore';
 import test from './test';
 import merge from 'deepmerge';
+import baseConfig from './config';
```
There's no advantage, I'm trying to use as few requires as possible so we can eventually make the switch to [this](https://socrata.atlassian.net/browse/EN-47234).

vickingan commented:
Updating babel doesn't require formatting changes. I added suggestions from jshint and cleaned up certain files as I was trying to use es6 modules with Node 14 earlier, so they are leftover remnants from [this](https://github.com/socrata-platform/geo-import/pull/113).
